src/pages/dev/general-message-passing/cosmos-gmp.mdx

```diff
@@ -1,29 +1,30 @@
----
-templateEngineOverride: md
----
-import { Callout } from "/src/components/callout"
+import { Callout } from "/src/components/callout";
 
 # Cosmos General Message Passing
 
 Axelar has expanded General Message Passing (GMP) to support Cosmos blockchains. With Axelar, you can now send and receive messages on EVM chains and Cosmos chains. Messages sent to Cosmos chains can be received by [CosmWasm smart contracts](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/tree/main/cosmwasm-integration/) (on blockchains with CosmWasm support), or those messages can be received [natively at the consensus layer as part Go code](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/tree/main/native-integration).
 
+Cosmos GMP works by sending and receiving through [IBC's memo field](https://medium.com/the-interchain-foundation/moving-beyond-simple-token-transfers-d42b2b1dc29b). Cosmos chains looking to support GMP should [integrate the appropriate middleware](#chain-support) and [verify message source](#establishing-a-path-of-trust).
 
 <Callout emoji="💡">
-To see full examples of sending and receiving messages in all cases, check out our full [Cosmos GMP Example Project](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/).
+  To see full examples of sending and receiving messages in all cases, check out
+  our full [Cosmos GMP Example
+  Project](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/).
 </Callout>
 
 ## Messages from EVM to CosmWasm
-To send a message from an EVM chain to a Cosmos CosmWasm contract, the payload must be encoded before calling the [standard](./gmp-messages) `payNativeGasForContractCall` and `callContract` methods (or [token-including equivalents](./gmp-tokens-with-messages)). You must also specify the destination method you are calling.
 
+To send a message from an EVM chain to a Cosmos CosmWasm contract, the payload must be encoded before calling the [standard](./gmp-messages) `payNativeGasForContractCall` and `callContract` methods (or [token-including equivalents](./gmp-tokens-with-messages)). You must also specify the destination method you are calling.
 
 The payload must specify:
 
-1. **string** - The method name of the CosmWasm contract to call
-1. **string[]** - The names of all arguments
-1. **string[]** - the types of all arguments
-1. **bytes** - the values of all arguments
+1. `string` - The method name of the CosmWasm contract to call
+1. `string[]` - The names of all arguments
+1. `string[]` - the types of all arguments
+1. `bytes` - the values of all arguments
+
+### Multi-send Example encoding
 
-### MultiSend Example encoding
 This example encodes a string array of recipients, as you can see in our [example of sending tokens and recipients](https://github.com/axelarnetwork/axelar-examples/blob/main/examples/evm/call-contract-with-token/DistributionExecutable.sol).
```

Comment:
Is this saying that the developer can look at the GitHub example of encoding payload recipients?

Reply:
yes

> Is this saying that the developer can look at the GitHub example of encoding payload recipients?